### ISSUE TITLE
Noisy tests

### ISSUE THREAD

**Issue by EmmanuelMess** [uprotocol/platform-java#2]
Test what happens when errors in transmission change the message.

**Comment by velitasali** [comment]
Yes, those tests are missing. I will write them as soon as I have time.

**Comment by velitasali** [comment]
I have implemented the basic structure and added test scenarios for that. I will cover most of the classes and use cases when I am done. That should also help improve the architecture of the API.
